temporal-sdk/src/main/java/io/temporal/internal/client/RootScheduleClientInvoker.java

```diff
@@ -101,12 +101,11 @@ public void createSchedule(CreateScheduleInput input) {
 
     try {
       genericClient.createSchedule(request.build());
-    } catch (Exception e) {
-      if (e instanceof StatusRuntimeException) {
-        StatusRuntimeException sre = (StatusRuntimeException) e;
-        if (Status.Code.ALREADY_EXISTS.equals(sre.getStatus().getCode())) {
-          throw new ScheduleAlreadyRunningException(sre);
-        }
+    } catch (StatusRuntimeException e) {
+      if (Status.Code.ALREADY_EXISTS.equals(e.getStatus().getCode())) {
+        throw new ScheduleAlreadyRunningException(e);
+      } else {
+        throw e;
```

Comment:
This should be wrapped in some subclass of `TemporalException` I am not sure if any current one fit this case, so we should probably add one

Reply:
I wonder if it's a general and separate issue to wrap all non-specialized schedule client exceptions.

Reply:
Yeah we could tackle it in separate issue, but I wouldn't release this fix until that issue is also resolved.

Reply:
Changes made 😁 but not sure about the error message